src/runtime_src/tools/xclbinutil/SectionIPLayout.cxx

```diff
@@ -119,6 +119,75 @@ SectionIPLayout::getIPControlType(std::string& _sIPControlType) const
   throw std::runtime_error(errMsg);
 }
 
+const std::string
+SectionIPLayout::getFunctionalStr(PS_FUNCTIONAL eFunctional) const
+{
+  switch (eFunctional) {
+    case FC_DPU:
+      return "DPU";
+    case FC_PREPOST:
+      return "PrePost";
+    default:
+      return (boost::format("UNKNOWN (%d)") % static_cast<unsigned int>(eFunctional)).str();
```

Comment:
In this case, an empty string should be returned.

Reply:
code refactored